autopush/web/validation.py

```diff
@@ -74,14 +74,19 @@ def _decorator(self, func):
         def wrapper(request_handler, *args, **kwargs):
             # Wrap the handler in @cyclone.web.synchronous
             request_handler._auto_finish = False
-
+            http_context = dict(
+                method=request_handler.request.method,
+                path=request_handler.request.path,
+                remote_ip=request_handler.request.remote_ip,
+            )
             d = deferToThread(self._validate_request, request_handler)
             d.addCallback(self._call_func, func, request_handler, *args,
                           **kwargs)
             d.addErrback(request_handler._overload_err)
             d.addErrback(request_handler._boto_err)
             d.addErrback(request_handler._validation_err)
-            d.addErrback(request_handler._response_err)
+            d.addErrback(request_handler._response_err,
+                         http_context=http_context)
```

Comment:
Why not just set the http context on the request_handler during its initialization? That way we always have it handy to pass in.